v1.1/performance-best-practices-overview.md

```diff
@@ -0,0 +1,196 @@
+
+## Unique ID Best Practices
+
+The common practice to generate unique IDs includes using transactions with roundtrip `SELECT` to generate monotonically-increasing unique IDs by incrementing an `INT` variable, or generate random unique IDs by using `SERIAL` variables. These common approaches of generating unique IDs hamper performance when used with a distributed database like CockroachDB, because the process of generating unique IDs using these approaches is serial, and CockroachDB improves performance by parallelizing processes whenever possible. The best practice to generate unique IDs with CockroachDB is to use `UUID`. Because the unique IDs generated using `UUID` are random, the process can be parallelized, thus improving performance.
+
+### Use `UUID` to Generate Unique IDs
+
+{% include faq/auto-generate-unique-ids_v1.1.html %}
+
+### Use `INSERT` with the `RETURNING` Clause to Generate Unique IDs
+
+If, for some reason, you cannot use `UUID` to generate the unique IDs, then you might resort to the common practice to generate unique IDs is to use roundtrip `SELECT` in a transaction. Instead, for improved performance, [use the `RETURNING` clause with the `INSERT` statement](insert.html#insert-and-return-values) instead.
+
+#### Generate Monotonically-Increasing Unique IDs
+
+Suppose the table schema is as follows:
+
+~~~ sql
+> CREATE TABLE X (
+ID1 INT,
+ID2 INT,
+ID3 INT DEFAULT 1,
+PRIMARY KEY (ID1,ID2));
+~~~
+
+The common approach to generate monotonically-increasing Unique IDs is a transaction using the `SELECT` statement:
+
+~~~ sql
+> BEGIN;
+
+> INSERT INTO X VALUES (1,1,1)
+
+> ON CONFLICT (ID1,ID2)
+
+> DO UPDATE SET ID3=X.ID3+1;
+
+> SELECT * FROM X WHERE ID1=1 AND ID2=1;
+
+> COMMIT;
+~~~
+
+However, the performance best practice is to use a `RETURNING` clause with `INSERT` instead of the transaction:
+
+~~~ sql
+> INSERT INTO X VALUES (1,1,1),(2,2,2),(3,3,3)
+ON CONFLICT (ID1,ID2)
+DO UPDATE SET ID3=X.ID3 + 1
+RETURNING ID1,ID2,ID3;
+~~~
+
+#### Generate Random Unique IDs
+
+Suppose the table schema is as follows:
+
+~~~ sql
+> CREATE TABLE X (
+ID1 INT,
+ID2 INT,
+ID3 SERIAL,
+PRIMARY KEY (ID1,ID2));
+~~~
+
+The common approach to generate random Unique IDs is a transaction using the `SELECT` statement:
+
+~~~ sql
+> BEGIN;
+
+> INSERT INTO X VALUES (1,1);
+
+> SELECT * FROM X WHERE ID1=1 AND ID2=1;
+
+> COMMIT;
+~~~
+
+However, the performance best practice is to use a `RETURNING` clause with `INSERT` instead of the transaction:
+
+~~~ sql
+> INSERT INTO X VALUES (1,1),(2,2),(3,3)
+RETURNING ID1,ID2,ID3;
+~~~
+
+## Indexes Best Practices
+
+### Use Secondary Indexes
+
+You can improve the performance of queries using columns besides the primary key with secondary indexes. You can create them:
```

Comment:
`using columns besides the primary key` is a bit ambiguous. Do we mean: `using columns not in the primary key`?

Reply:
Also, for added clarity, I'd rearrange the sentence a bit:
> You can use secondary indexes to improve the performance of queries using columns not in a table's primary key. You can create them:

I know this language is copied from the `indexes.md` file, so can you make the updates there as well?